### PR TITLE
Feature/ui tracker tags tab integration

### PR DESCRIPTION
**Pull request by astewes**
-  Fetches tag list from the backend
-  Enables user to enter a list of preferred tags
-  Enables user to delete a preferred tag when it is not tagged to any entities
-  Enables user to promote a user tag to a preferred tag and demote a preferred tag to a user tag
-  Adds pagination to tables for both user and preferred tags when either list exceeds 10 tags

Note: Drag N Drop uploading of tags via CSV/TXT file will be handled in a follow-up PR

tags

Adding a list of preferred tags (w/validation):
validate

Scenario where there are invalid tags:
invalid

### CODE REVIEW

File: cdap-ui/app/features/tracker/controllers/tags-ctrl.js [hunk 1]
```diff
@@ -20,67 +20,182 @@ class TrackerTagsController{
...
+          });
+        });
+
+        angular.forEach((response.userTags), (count, tag) => {
+          this.tags.userTags.push({
+            name: tag,
+            count: count
+          });
+        });
+        console.log('this.tags.preferredTags', this.tags.preferredTags);
```
Comment (ajainarayanan) [inline cdap-ui/app/features/tracker/controllers/tags-ctrl.js]: Remove?

File: cdap-ui/app/features/tracker/controllers/tags-ctrl.js [hunk 2]
```diff
@@ -20,67 +20,215 @@ class TrackerTagsController{
...
+        this.tags = {
+          preferredTags: [],
+          userTags: []
+        };
+
+        angular.forEach(response.preferredTags, (count, tag) => {
+          this.tags.preferredTags.push({
+            name: tag,
+            count: count,
+            empty: false
```
Comment (elfenheart) [inline cdap-ui/app/features/tracker/controllers/tags-ctrl.js:47]: i don't think we need this property anymore?

File: cdap-ui/app/features/tracker/controllers/tags-ctrl.js [hunk 3]
```diff
@@ -20,67 +20,215 @@ class TrackerTagsController{
...
-        },
-        {
-          'key': 'tag5',
-          'total_entity': 28
+
+        angular.forEach((response.userTags), (count, tag) => {
+          this.tags.userTags.push({
+            name: tag,
+            count: count,
+            empty: false
```
Comment (elfenheart) [inline cdap-ui/app/features/tracker/controllers/tags-ctrl.js:63]: ditto

File: cdap-ui/app/features/tracker/services/my-tracker-api.js [hunk 4]
```diff
@@ -29,7 +29,12 @@ function myTrackerApi(myCdapUrl, $resource, myAuth, myHelpers, UI_CONFIG) {
...
       baseQueryPath = '/data/explore/queries/:handle',
-      truthMeterPath = '/namespaces/:namespace/apps/' + UI_CONFIG.tracker.appId + '/services/' + UI_CONFIG.tracker.serviceId + '/methods/v1/tracker-meter';
+      truthMeterPath = '/namespaces/:namespace/apps/' + UI_CONFIG.tracker.appId + '/services/' + UI_CONFIG.tracker.serviceId + '/methods/v1/tracker-meter',
+      tagsPath = '/namespaces/:namespace/apps/' + UI_CONFIG.tracker.appId + '/services/' + UI_CONFIG.tracker.serviceId + '/methods/v1/tags',
+      demotePreferredTagsPath = '/namespaces/:namespace/apps/' + UI_CONFIG.tracker.appId + '/services/' + UI_CONFIG.tracker.serviceId + '/methods/v1/tags/demote',
+      promoteUserTagsPath = '/namespaces/:namespace/apps/' + UI_CONFIG.tracker.appId + '/services/' + UI_CONFIG.tracker.serviceId + '/methods/v1/tags/promote',
+      validateTagsPath = '/namespaces/:namespace/apps/' + UI_CONFIG.tracker.appId + '/services/' + UI_CONFIG.tracker.serviceId + '/methods/v1/tags/validate',
+      deletePreferredTagsPath = '/namespaces/:namespace/apps/' + UI_CONFIG.tracker.appId + '/services/' + UI_CONFIG.tracker.serviceId + '/methods/v1/tags/preferred';
```
Comment (elfenheart) [inline cdap-ui/app/features/tracker/services/my-tracker-api.js:37]: Can we clean up these paths? Use the tagsPath base, and add whatever path you need

**Pull request closed without merging**